sympy/solvers/solveset.py

```diff
@@ -365,27 +389,19 @@ def _solve_as_rational(f, symbol, domain):
 
 def _solve_trig(f, symbol, domain):
     """ Helper to solve trigonometric equations """
-    f = trigsimp(f)
-    f_original = f
+    if _is_function_class_equation(TrigonometricFunction, f, symbol):
```

Comment:
Changes regarding [XFAIL testcases of Hyperbolic Functions ](https://github.com/sympy/sympy/pull/10764/files#diff-85baa04bbf4e1dfd9128782738e45424R577)